tests/queries/0_stateless/02174_cte_scalar_cache.sql

```diff
@@ -0,0 +1,48 @@
+WITH
+    ( SELECT sleep(0.0001) FROM system.one ) as a1,
+    ( SELECT sleep(0.0001) FROM system.one ) as a2,
+    ( SELECT sleep(0.0001) FROM system.one ) as a3,
+    ( SELECT sleep(0.0001) FROM system.one ) as a4,
+    ( SELECT sleep(0.0001) FROM system.one ) as a5
```

Comment:
I had no idea that `enable_global_with_statement` can actually add prev `with` result to the subquery inside `with` ...

Reply:
Yep, the first query makes 5 sleep calls (before it was 9 IIRC) and the second one, with `enable_global_with_statement=0` only one.